app/js/questions_controller.js

```diff
@@ -114,6 +109,17 @@
           show_more = true;
         }
 
+        var date_format = new Intl.DateTimeFormat(navigator.language, {
+          weekday: 'long', month: 'long', day: 'numeric'
+        });
```

Comment:
You're right we need to pay attention to this for earlier versions. We're not going to use the l10n library for translations so that's why I went with this API.
I'll open a bug and we'll see what we'll do in that bug. Maybe we can use a polyfill.